framework/src/org/checkerframework/framework/test/TestConfigurationBuilder.java

```diff
@@ -0,0 +1,363 @@
+package org.checkerframework.framework.test;
+
+import org.checkerframework.framework.util.PluginUtil;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class TestConfigurationBuilder {
+
+
+    private List<File> diagnosticFiles;
+    private List<File> testSourceFiles;
+    private Set<String> processors;
+    private SimpleOptionMap options;
+
+    private boolean shouldEmitDebugInfo;
+
+    public TestConfigurationBuilder() {
```

Comment:
I did not plan on forcing people to use the static methods.  It can be used either way.  I just added those methods in order to make it easy to cover the most common case.  I can move them up top, however.